tree/ntuple/v7/src/RNTupleSerialize.cxx

```diff
@@ -53,7 +62,9 @@ std::uint32_t SerializeField(const ROOT::Experimental::RFieldDescriptor &fieldDe
    pos += RNTupleSerializer::SerializeUInt32(fieldDesc.GetFieldVersion(), *where);
    pos += RNTupleSerializer::SerializeUInt32(fieldDesc.GetTypeVersion(), *where);
    pos += RNTupleSerializer::SerializeUInt32(onDiskParentId, *where);
-   pos += RNTupleSerializer::SerializeFieldStructure(fieldDesc.GetStructure(), *where);
+   std::uint32_t result;
+   UNWRAP_OR_RETURN(result, RNTupleSerializer::SerializeFieldStructure(fieldDesc.GetStructure(), *where));
+   pos += result;
```

Comment:
For example
```suggestion
   auto result = RNTupleSerializer::SerializeFieldStructure(fieldDesc.GetStructure(), *where)
   if (!result)
      return R__FORWARD_ERROR(result);
   pos += result.Unwrap();
```
is not much longer and IMHO much more readable.

Reply:
I actually find the macro much "cleaner" since it neatly wraps a lot of repeated line noise and it slightly reduces the chance of copypaste errors. It makes it very clear that in that place there is error forwarding involved and lets the reader split up easily the `if`s that are some more specific logic from those that are just error propagation.

I'd like to hear the other reviewers' opinion as well so we can better come up to an agreement.

Reply:
```
   if (auto result = RNTupleSerializer::SerializeFieldStructure(fieldDesc.GetStructure(), *where))
       pos += result.Unwrap();
   else
      return R__FORWARD_ERROR(result);
```
might be another option.